website/server/libs/groupTasks.js

```diff
@@ -0,0 +1,46 @@
+import * as Tasks from '../models/task';
+
+async function handleSharedCompletion (groupMemberTask) {
+  let masterTask = await Tasks.Task.findOne({
+    _id: groupMemberTask.group.taskId,
+  }).exec();
+
+  if (!masterTask || !masterTask.group || masterTask.type !== 'todo') return undefined;
+
+  if (masterTask.group.sharedCompletion === 'singleCompletion') {
+    await Tasks.Task.deleteMany({
```

Comment:
let's make sure all these queries are backed by indexes

Reply:
https://github.com/HabitRPG/habitica/blob/48bcc1e968264e9e41d7074b14dc33587624a95d/migrations/docs/mongo-indexes.md#tasks should cover it, yes? `group.taskId` and `group.userId`

Reply:
I suppose `group.approval.approved` isn't indexed. We can add that if you think it'd be valuable! But I'm only querying that in conjunction with `group.taskId` so I thought it'd be a pretty narrow pool to begin with.